__tests__/any-of-labels.spec.ts

```diff
@@ -0,0 +1,106 @@
+import {Issue} from '../src/classes/issue';
+import {IssuesProcessor} from '../src/classes/issues-processor';
+import {IIssuesProcessorOptions} from '../src/interfaces/issues-processor-options';
+import {DefaultProcessorOptions} from './constants/default-processor-options';
+import {generateIssue} from './functions/generate-issue';
+
+describe('any-of-labels option', () => {
+  test('should do nothing when not set', async () => {
+    const sut = new IssuesProcessorBuilder()
+      .emptyAnyOfLabels()
+      .issues([{labels: [{name: 'some-label'}]}])
+      .build();
+
+    await sut.processIssues();
+
+    expect(sut.staleIssues).toHaveLength(1);
+  });
+
+  test('should skip it when none of the issue labels match', async () => {
+    const sut = new IssuesProcessorBuilder()
+      .anyOfLabels('skip-this-issue,and-this-one')
+      .issues([{labels: [{name: 'some-label'}, {name: 'some-other-label'}]}])
+      .build();
+
+    await sut.processIssues();
+
+    expect(sut.staleIssues).toHaveLength(0);
+  });
+
+  test('should skip it when the issue has no labels', async () => {
+    const sut = new IssuesProcessorBuilder()
+      .anyOfLabels('skip-this-issue,and-this-one')
+      .issues([{labels: []}])
+      .build();
+
+    await sut.processIssues();
+
+    expect(sut.staleIssues).toHaveLength(0);
+  });
+
+  test('should process it when one of the issue labels match', async () => {
+    const sut = new IssuesProcessorBuilder()
+      .anyOfLabels('skip-this-issue,and-this-one')
+      .issues([{labels: [{name: 'some-label'}, {name: 'skip-this-issue'}]}])
+      .build();
+
+    await sut.processIssues();
+
+    expect(sut.staleIssues).toHaveLength(1);
+  });
+
+  test('should process it when all the issue labels match', async () => {
+    const sut = new IssuesProcessorBuilder()
+      .anyOfLabels('skip-this-issue,and-this-one')
+      .issues([{labels: [{name: 'and-this-one'}, {name: 'skip-this-issue'}]}])
+      .build();
+
+    await sut.processIssues();
+
+    expect(sut.staleIssues).toHaveLength(1);
+  });
+});
+
+class IssuesProcessorBuilder {
+  private _options: IIssuesProcessorOptions;
+  private _issues: Issue[];
+
+  constructor() {
+    this._options = {...DefaultProcessorOptions};
```

Comment:
Why not assign the properties on creation to reduce the class size instead of using a constructor?

Reply:
Ah, a OO purist would say the constructor is the place to do it! lol. But really, I'm in the habit of doing it like this for more practical reasons: 1) if you add some more fields, and some of them are initialized via a constructor arg, and others are initialized inline where they are declared, that could get a bit disorganized, 2) it gives a good indication for potential refactorings, e.g. those default options could be injected, instead of imported (imagine we have some other predefined defaults, for example).
But it's just a matter of preference, really. I'm not opposed to getting rid of the constructor 😀.